### PR TITLE
Upgrade to a proper async worker for sync call, fixes #138

### LINKED ISSUE #138: env.sync() blocks the micro event queue (promise callbacks) in Node when other timeouts are in the queue
This is a minimal test case to reproduce the issue:
```
const { Env } = require('node-lmdb');
let env = new Env();
env.open({path: 'test'}); // some path for the db
new Promise(resolve => {
	env.sync(() => {
		console.log('synced');
		resolve();
	})
}).then(() => {
	// this should execute immediately after it is synced
	console.log('then');
});
setTimeout(() => {
	console.log('longer timeout');
}, 10000);
```
Expected output:
```
synced
then
longer timeout
```
Actual output:
```
synced
longer timeout
then
```

### PR DESCRIPTION
**Pull request by kriszyp**

### CODE REVIEW

File: src/env.cpp [hunk 1]
```diff
@@ -98,6 +98,36 @@ int applyUint32Setting(int (*f)(MDB_env *, T), MDB_env* e, Local<Object> options
...
 }
 
+class SyncWorker : public Nan::AsyncWorker {
+  public:
+    SyncWorker(MDB_env* env, Nan::Callback *callback)
+      : Nan::AsyncWorker(callback), env(env) {}
+
+/*    ~SyncWorker() {
+      delete env;
+    }*/
```
Comment (Venemo) [inline src/env.cpp]: You deleted the same lines in your previous commit, I assume these are here by mistake?

File: src/env.cpp [hunk 2]
```diff
@@ -307,37 +337,16 @@ NAN_METHOD(EnvWrap::sync) {
         return Nan::ThrowError("The environment is already closed.");
     }
 
-    Local<Function> callback = info[0].As<Function>();
-
-    EnvSyncData *d = new EnvSyncData;
```
Comment (Venemo) [inline src/env.cpp]: This was the only place where `EnvSyncData` was ever used, can you please delete it now that it's not used anymore?

File: src/env.cpp (hunk 1 above)
Comment (kriszyp) [reply src/env.cpp]: I think I deleted it in the later/2nd commit. It makes sense that isn't needed, right?

**Comment by kriszyp** [comment]
I added the unit test for this issue (which fails before fix and passes afterwards for me), and cleaned up EnvSyncData. Let me know if there is anything else you want me to do. Thanks!

**Comment by Venemo** [comment]
Thanks. Merged now.

**Pull request merged**